### PR TITLE
refactor(server): use helper.arrayRemove in appropriate places

### PR DESCRIPTION
**Pull request by lusarz**

### CODE REVIEW

File: lib/browser.js [hunk 1]
```diff
@@ -143,7 +143,7 @@ class Browser {
   }
 
   onDisconnect (disconnectedSocket) {
-    this.activeSockets.splice(this.activeSockets.indexOf(disconnectedSocket), 1)
```
Comment (lusarz) [inline lib/browser.js]: Hmm, it looked incorrect. When `disconnectedSocket` is not in `this.activeSockets` array then `this.activeSockets.indexOf(disconnectedSocket)` returns `-1`.

    array.splice(-1, 1)

remove last element of array.

I'll analyze unit tests

Comment (lusarz) [reply lib/browser.js]: `onDisconnect` method was called in unit test with two params instead of one, I fixed it

Comment (johnjbarton) [reply lib/browser.js]: The disconnect event sends a reason:
https://socket.io/docs/client-api/#Event-%E2%80%98disconnect%E2%80%99
But don't pass it through our handler.  Can we pass it in and append it to the this.disconnect() call? 

We see rare disconnect fails and better logging may help sort out why.

File: lib/browser.js [hunk 2]
```diff
@@ -142,24 +142,24 @@ class Browser {
...
+    helper.arrayRemove(this.activeSockets, disconnectedSocket)
 
     if (this.activeSockets.length) {
       this.log.debug('Disconnected %s, still have %s', disconnectedSocket.id, this.getActiveSocketsIds())
       return
     }
 
     if (this.state === CONNECTED) {
-      this.disconnect('client disconnected from CONNECTED state')
+      this.disconnect(`client disconnected from CONNECTED state (${reason})`)
```
Comment (lusarz) [inline lib/browser.js]: @johnjbarton 
Please take a look if it's good and enough way of passing reason into disconnect.

**Pull request merged**